service/service_construction.go

```diff
@@ -276,14 +299,64 @@ func (s ConstructionService) ConstructionParse(
 		tx.From = msg.From().Hex()
 	}
 
+	metadata := &parseMetadata{
+		Nonce:    tx.Nonce,
+		GasPrice: tx.GasPrice,
+		GasLimit: tx.GasLimit,
+		ChainID:  tx.ChainID,
+	}
+	metaMap, err := marshalJSONMap(metadata)
+	if err != nil {
+		return nil, wrapError(errInternalError, err)
+	}
+	var ops []*types.Operation
+	var checkFrom *string
+	var wrappedErr *types.Error
+
+	if len(tx.Data) != 0 {
+		unwrapMethodID := getMethodID(unwrapFnSignature)
+		if hexutil.Encode(tx.Data[:4]) == hexutil.Encode(unwrapMethodID) {
```

Comment:
We should do a switch here and error if not unwrap or transfer. This is too casual.

Reply:
#188